interpreter.asm

```diff
@@ -256,61 +246,38 @@ print_word_not_found_error ; ( caddr u -- )
 
     +BACKLINK "'", 1
     jsr PARSE_NAME
-    jsr TWODUP
-    jsr FIND_NAME
-    inx
-    lda MSB-1,x
-    beq print_word_not_found_error
-+   ldy LSB-1, x
-    sty LSB, x
-    sta MSB, x
-    sty LSB+1, x
-    sta MSB+1, x
-    jsr TO_XT
-    jsr SWAP
-    jsr GET_IMMED
-    inx
-    rts
-
-    +BACKLINK "find", 4
-FIND ; ( xt -1 | xt 1 | caddr 0 )
-    jsr DUP
-    jsr TO_R
-    jsr COUNT
-    jsr FIND_NAME
-    lda MSB, x
-    beq +
-    jsr DUP
-    jsr TO_XT
-    jsr SWAP
-    jsr GET_IMMED
-    jsr R_TO
-    inx
-    rts
+    lda LSB,x
+    pha
+    jsr FIND
+    lda LSB,x
+    bne +
+    pla
+    sta LSB,x
+    jmp print_word_not_found_error
 +   inx
-    jsr R_TO
-    jmp ZERO
+    pla
+    rts
 
 FIND_BUFFER = $33c
 FIND_BUFFER_SIZE = 31
 
-    +BACKLINK "find-name", 9
-FIND_NAME ; ( caddr u -- nt | 0 )
-    inx
-    lda LSB-1,x
+    +BACKLINK "find", 4
+    jsr COUNT
+FIND ; ( caddr u -- xt -1 | xt 1 | caddr 0 )
```

Comment:
This is a kludge, allowing the local code to use a non-counted string.  It's also a bug according to the standard, as this code does not return the caddr but the saddr when failing.

No code in the base relies on the standard behaviour, but that could be fixed.

Reply:
Thoughts @jkotlinski ?

Reply:
Christmas vacations ended and so I do not have so much spare time or brain cycles anymore :-)  I still have to delve into this PR to understand what it is about.

In general, I think it is strongly preferable to be standard compliant when possible.